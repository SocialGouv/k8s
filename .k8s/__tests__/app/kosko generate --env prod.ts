//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing"
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env"

const name = "myapp"

jest.setTimeout(1000 * 60)

test("kosko generate --prod", async () => {
  expect(
    await getEnvManifests("prod", "'!(_*)'", {
      ...project(name).prod,
      SOCIALGOUV_CONFIG_PATH: __dirname + "/config.json",
    })
  ).toMatchSnapshot()
})
