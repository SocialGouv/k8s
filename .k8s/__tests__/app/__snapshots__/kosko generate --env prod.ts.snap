// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app: kosko generate --prod 1`] = `
"---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: prod2
    app.gitlab.com/env.name: prod2
  labels:
    app: myapp
    application: myapp
    owner: myapp
    team: myapp
  name: myapp
  namespace: myapp
spec:
  replicas: 1
  selector:
    matchLabels:
      app: myapp
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-myapp
        app.gitlab.com/env: prod2
        app.gitlab.com/env.name: prod2
      labels:
        app: myapp
        application: myapp
        owner: myapp
        team: myapp
    spec:
      containers:
        - image: harbor.fabrique.social.gouv.fr/myapp/myapp:1.2.3
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: myapp
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 200m
              memory: 256Mi
            requests:
              cpu: 50m
              memory: 128Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          env:
            - name: APP_BASE_URL
              value: https://myapp.fabrique.social.gouv.fr
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: myapp
    application: myapp
    owner: myapp
    team: myapp
  name: myapp
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: prod2
    app.gitlab.com/env.name: prod2
  namespace: myapp
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: myapp
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
    kubernetes.io/tls-acme: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: prod2
    app.gitlab.com/env.name: prod2
  labels:
    app: myapp
    application: myapp
    owner: myapp
    team: myapp
  name: myapp
  namespace: myapp
spec:
  rules:
    - host: myapp.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: myapp
              servicePort: 80
            path: /
  tls:
    - hosts:
        - myapp.fabrique.social.gouv.fr
      secretName: myapp-crt
"
`;
